### ISSUE TITLE
"Downloading media failed: http error: HTTP status client error (404 Not Found) for url" for old URLs

### ISSUE THREAD

**Issue by Suika** [miyachan/torako#1]
TL;DR Torako uses the original preview_op/media name to an image that already exists in the DB but not on disk. But should actually use the preview_op/media from the 4ch JSON response and save to original preview_op/media.
Using the old information causes Torako to fail re-downloading the file. Noticed it becasue Torako was erroring out on a lot of files that I haven'T moved to my destination, yet. But had that file somewhere else.

```
 2021-02-02T23:06:42.928Z ERROR torako::storage::asagi           > Downloading media failed: http error: HTTP status client error (404 Not Found) for url (https://i.4cdn.org/fit/1607387438265.png)
```
![grafik](https://user-images.githubusercontent.com/2320837/106676171-4ddd5d80-65ae-11eb-9235-7315652963ab.png)
![grafik](https://user-images.githubusercontent.com/2320837/106676187-533aa800-65ae-11eb-96fc-575391cf8d60.png)

**Comment by Suika** [comment]
Also, it might be needed by some archivers to have the ability to disable re-downloading. In case they move a portion of the data to a remote location.
Basically, so that the the missing files are not re-downloaded.

**Comment by miyachan** [comment]
> Also, it might be needed by some archivers to have the ability to disable re-downloading. In case they move a portion of the data to a remote location.

Is this fine as a global option or is it worth it to have it at the board level?

**Comment by Suika** [comment]
Good question, but I don't know. It depends on the person using it.
I am fine with global option, but maybe people want to have it on board level. Who knows.
Ask around in the chat.

**Comment by miyachan** [comment]
In the process of trying to figure out how to provide a `redownload_if_missing` option, I don't think it's possible to do so currently. The problem is there's no way to tell if it's missing because we have *never* downloaded it or if its missing because someone deleted it with the current Asagi schema. It would require adding a new column to the Asagi schema when an image is successfully downloaded and using that to inform us if we are redownloading or not.

The primary issue for why this was uploaded is fixed in 0.11.2
